contracts/service-provider-directory/src/error.rs

```diff
@@ -0,0 +1,28 @@
+use cosmwasm_std::{Addr, StdError};
+use thiserror::Error;
+
+use crate::state::ServiceId;
+
+#[derive(Error, Debug, PartialEq)]
+pub enum ContractError {
+    #[error("{0}")]
+    Std(#[from] StdError),
+
+    #[error("service not found: {service_id}")]
+    NotFound { service_id: ServiceId },
+
+    #[error("{sender} is not owner of service")]
+    Unauthorized { sender: Addr },
+
+    #[error("deposit required to announce service")]
+    DepositRequired { source: cw_utils::PaymentError },
+
+    #[error("insufficiant deposit: {funds}, required: {deposit_required}")]
+    InsufficientDeposit {
+        funds: cosmwasm_std::Uint128,
```

Comment:
use the same type everywhere because it will be confusing otherwise (i.e. either `Uint128` or `Coin`)